python/tank/templatekey.py

```diff
@@ -256,6 +272,124 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        default=None,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param default: Default value for this field. Acceptable values are
+                            - None
+                            - a datetime.datetime object
+                            - a string formatted according to the format_spec
+                            - utc_now, which means the current time in the UTC timezone will be used
+                              as the default value.
+                            - now, which means the current time in the local timezone will be used
+                              as the default value.
+        :param format_spec: Specification for formating when casting to/from a string.
+                            The format follows the convention of strftime and strptime. The
+                            default value is "%Y-%m-%d-%H-%M-%S". Given June 24th, 2015 at
+                            9:20:30 PM, this will yield 2015-06-24-21-20-30
+        """
+
+        # Can't use __repr__ because of a chicken and egg problem. The base class validates the
+        # default value, so format_spec needs to be set first. But if I am testing format_spec
+        # before calling the base class, then repr will crash since self.name won't have been set
+        # yet.
+        if isinstance(format_spec, basestring) is False:
+            raise TankError("format_spec for <Sgtk TimestampKey %s> is not of type string: %s" %
+                            (name, format_spec.__class__.__name__))
+        self.format_spec = format_spec
+
+        if isinstance(default, basestring):
+            # if the user passes in now or utc, we'll generate the current time as the default time.
+            if default.lower() == "now":
+                self._default_to_utc = False
+                default = self.__get_current_time
```

Comment:
Is it simpler to have two methods `self.__get_current_time` and `self.__get_current_utc_time`?